src/main/java/io/reactivex/internal/subscriptions/DeferredScalarSubscription.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.subscriptions;
+
+import org.reactivestreams.Subscriber;
+
+/**
+ * A subscription that signals a single value eventually.
+ * <p>
+ * Note that the class leaks all methods of {@link java.util.concurrent.atomic.AtomicLong}.
+ * Use {@link #complete(Object)} to signal the single value.
+ * @param <T> the value type
+ */
+public class DeferredScalarSubscription<T> extends BasicQueueSubscription<T> {
```

Comment:
New tests will be added before RC1 based on missing tests from 1.x and code coverage.